File: python/agno_platform/privacy/sensitive_fields.py
"""Sensitive field classifications shared with the TypeScript contracts library."""

SENSITIVE_FIELD_KEY_FRAGMENTS = (
    "authorization",
    "cookie",
    "token",
    "secret",
    "password",
    "prompt",
    "patientaddress",
    "toolpayload",
)


def is_sensitive_field_key(key: str) -> bool:
    normalized_key = "".join(character for character in key.lower() if character.isalnum())
    return any(fragment in normalized_key for fragment in SENSITIVE_FIELD_KEY_FRAGMENTS)

File: libs/contracts/src/lib/privacy/sensitive-fields.ts
/**
 * Case-insensitive key fragments that must never be emitted to logs or traces.
 * This list is mirrored by python/agno_platform/privacy/sensitive_fields.py.
 */
export const sensitiveFieldKeyFragments = [
  'authorization',
  'cookie',
  'token',
  'secret',
  'password',
  'prompt',
  'patientaddress',
  'toolpayload',
] as const;

export function isSensitiveFieldKey(key: string): boolean {
  const normalizedKey = key.toLowerCase().replace(/[^a-z0-9]/g, '');
  return sensitiveFieldKeyFragments.some((fragment) => normalizedKey.includes(fragment));
}
